### PR TITLE
[SPARK-6204][SQL] GenerateProjection's equals should check length equality

### PR DESCRIPTION
**Pull request by viirya**
`GenerateProjection`'s `equals` now only checks column equality. It should also check length equality.

### CODE REVIEW

**Comment by yhuai** [comment]
Can you add a test?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateProjection.scala [hunk 1]
```diff
@@ -52,7 +52,7 @@ object GenerateProjection extends CodeGenerator[Seq[Expression], Projection] {
 
     val nullFunctions =
       q"""
-        private[this] var nullBits = new Array[Boolean](${expressions.size})
+        private[this] var nullBits = new Array[Boolean](this.length)
```
Comment (yhuai) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateProjection.scala]: Seems we do not really have to change this?

**Comment by marmbrus** [comment]
I don't think change is required.  The class check in the `match` for equality is looking for this exact instance of SpecificRow which will always have exactly the same length (because it must be generated from the same expressions).

Unless you have a test case that shows a failure (and in general it is good to always have one when filing a bug) I suggest we close this issue.

**Comment by viirya** [comment]
It would not get a failure. For now, the `equals` function will go back to call `super.equals` to test the equality if two rows have different lengths.

**Comment by viirya** [comment]
If you think this is not worth fixing. Let's close this pr.

**Pull request closed without merging**